### ISSUE TITLE
crashes for Safari browser

### ISSUE THREAD

**Issue by lishenyu16** [HerbertHe/for-editor-herb#10]
looks that somewhere you are using regular expression that causes it to crash on Safari browser: 

> SyntaxError: Invalid regular expressions: invalid group specifier name

**Comment by lishenyu16** [comment]
`                            /(?<=\+ )/.test(item) ? backDiff_1 += '<p class="for-md-diff-add">' + item + "</p>" : /(?<=\- )/.test(item) ? backDiff_1 += '<p class="for-md-diff-del">' + item + "</p>" : /(?<=\! )/.test(item) ? backDiff_1 += '<p class="for-md-diff-focus">' + item + "</p>" : /(?<=\# )/.test(item) ? backDiff_1 += '<p class="for-md-diff-ignore">' + item + "</p>" : backDiff_1 += "<p>" + item + "</p>";

`

**Comment by lishenyu16** [comment]
```javascript
                    if ("diff" === language) {
                        var backDiff_1 = "";
                        return code.split("\n").forEach(function(item) {
                            /(?<=\+ )/.test(item) ? backDiff_1 += '<p class="for-md-diff-add">' + item + "</p>" : /(?<=\- )/.test(item) ? backDiff_1 += '<p class="for-md-diff-del">' + item + "</p>" : /(?<=\! )/.test(item) ? backDiff_1 += '<p class="for-md-diff-focus">' + item + "</p>" : /(?<=\# )/.test(item) ? backDiff_1 += '<p class="for-md-diff-ignore">' + item + "</p>" : backDiff_1 += "<p>" + item + "</p>";
                        }), '<pre class="for-md-diff"><code>' + backDiff_1 + "</code></pre>";
                    }
```

**Comment by HerbertHe** [comment]
I have searched for the answer, the error happens in webpack packing, I will fix in the next version.

**Comment by HerbertHe** [comment]
I don't have macosx environment, if the trouble still here, plz let me know, thx

**Comment by lishenyu16** [comment]
> I don't have macosx environment, if the trouble still here, plz let me know, thx

if you have iphone, u can use iphone safari browser to test your [https://goer.icu/for-editor-herb/](url)

**Comment by HerbertHe** [comment]
> > I don't have macosx environment, if the trouble still here, plz let me know, thx
> 
> if you have iphone, u can use iphone safari browser to test your [https://goer.icu/for-editor-herb/](url)

Is the trouble still here? I use Android device often 😢

**Comment by lishenyu16** [comment]
> https://goer.icu/for-editor-herb/

yes the issue is still there on the demo page

**Comment by lishenyu16** [comment]
sorry I only tested your demo page, it's fixed now(at least is working fine now). I'll test more later.  Thanks

2.3.6 version is working fine now. Your demo page may need to be re-deployed.

**Comment by ivanandonov** [comment]
Apparently those regex that start with `?<=` are considered as errors.
I manage to make it work when i changed them with RegExp constructor for example:
`/(?<=\+ )/` to new `RegExp("(?<=\\+ )")` there are 6 of those in the code. If that webpack packing doesn't fix it you have another solution ;)

**Comment by HerbertHe** [comment]
> Apparently those regex that start with `?<=` are considered as errors.
> I manage to make it work when i changed them with RegExp constructor for example:
> `/(?<=\+ )/` to new `RegExp("(?<=\\+ )")` there are 6 of those in the code. If that webpack packing doesn't fix it you have another solution ;)

yes, I tried to use RegExp constructor to fix this bug in version 2.3.6, I will try to find more solutions

**Comment by HerbertHe** [comment]
I don't know if everything is ok now, I tried to rewrite RegExp
